src/main/kotlin/biokotlin/kmer/Kmer.kt

```diff
@@ -0,0 +1,155 @@
+package biokotlin.kmer
+
+import biokotlin.seq.NucSeq
+import biokotlin.seq.Seq
+
+/**
+ * Create a Kmer from a string
+ * Sequence may be fewer than 32 bases long, left will be padded with A's to equal 32 bases
+ * @param seq String representation of a nucleotide sequence
+ */
+fun Kmer(seq: String): Kmer {
+    if (seq.length > 32) {throw java.lang.IllegalArgumentException("Kmer must be less than or equal to 32 bases long")}
+
+    if( seq.contains(Regex("[^ACTGU]"))) {throw java.lang.IllegalArgumentException("Kmer may contain only A, C, T, G, U")}
+    var encoding = 0L
+    seq.forEach{ c -> encoding = (encoding shl 2) or (c.code.toLong() ushr 1 and 3L) }
+    return Kmer(encoding)
+}
+
+/**
+ * Create a Kmer from a NucSeq
+ * Sequence may be fewer than 32 bases long, left will be padded with A's to equal 32 bases
+ * @param seq NucSeq without ambiguous bases
+ */
+fun Kmer(seq: NucSeq): Kmer {
+    return Kmer(seq.seq())
+}
+
+/**
+ * 2-bit encoded representation of a short nucleotide sequence (32 bases or fewer)
+ * Encoding: A=0, C=1, T/U=2, G=3
+ * For kmers smaller than 32 bp, leftmost digits are padded with 0
+ * Ambiguous bases (eg. N) are not allowed
+ *
+ *
+ * ``` kotlin
+ * val kmer = Kmer(1432462)
+ * val kmerFromString = Kmer("UGCACUGCCA")
+ * val kmerFromSeq = Kmer(NucSeq("TCTCACCTACA")
+ *
+ * val kmerFromString = Kmer("NTACCGANNTCG") // throw error - ambiguous bases
+ * val kmerFromString = Kmer("TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT") // throw error - string too long
+ * ```
+ */
+@JvmInline
+value class Kmer(val encoding: Long): Comparable<Kmer> {
+
+    /** Follows comparisons of bits represented as unsigned long */
+    override operator fun compareTo(other: Kmer): Int {
+        return encoding.toULong().compareTo(other.encoding.toULong())
+    }
+
+    /**
+     * Converts 2-bit encoding to string
+     * By default, keeps any A padding at left
+     */
+    override fun toString(): String {
+        return toString(32)
+    }
+
+    /**
+     * Converts 2-bit encoding to string, trimming leftmost bits
+     * so the resulting sequence length is [kmerSize]
+     */
+    fun toString(kmerSize: Int): String {
+        var temp = encoding
+        return (0 until kmerSize).map{
+            val y = when ( temp and 3L) {
+                0L -> 'A'
```

Comment:
Are these codings defined by constants somewhere in BioKotlin? Seems like they should be to be consistent?